driver-sync/src/test/functional/com/mongodb/client/unified/UnifiedTest.java

```diff
@@ -449,7 +467,7 @@ private OperationResult executeOperation(final UnifiedTestContext context, final
                 case "abortTransaction":
                     return crudHelper.executeAbortTransaction(operation);
                 case "withTransaction":
-                    return crudHelper.executeWithTransaction(operation, (op, idx) -> assertOperation(context, op, idx));
+                    return crudHelper.executeWithTransaction(operation, (op, idx) -> assertOperationAndThrow(context, op, idx));
```

Comment:
This is necessary because, unlike a normal operation where we just assert the result (exceptional or not) and then move on to the next one, when an operation is executed in the body of a withTransaction callback, any exception thrown by that operation has to propagate out of the callback, so that it can trigger an abortTransaction instead of a commitTransaction.